Update release process documentation (#17461)
galaxyproject/galaxy: doc/source/dev/create_release.rst

Creating Galaxy Releases
========================

The "main" release process is an interactive checklist with instructions (see `Publication of Galaxy Release v 23.2 https://github.com/galaxyproject/galaxy/issues/16742>`_ for an example).

> This issue is generated via `make release-issue` at the end of the process.

Reviewer: ```suggestion
This issue is generated via `make release-issue`.
```
I think this is all we need here? I think "at the end of the process" may be a little confusing since we generate it at the beginning of the release process (and the last bullet points in the issue refer to the *next* release).

> The final result of the release process are
> - a new branch (release_YY.N) from which point releases are created
> - a tag pointing at the first commit of the branch (vYY.N)

Reviewer: ```suggestion
The final result of the release process are

- a new branch (release_YY.N) from which point releases are created
- a tag pointing at the first commit of the branch (vYY.N)
```
Otherwise it won't render as a list.

I'm not sure about generating a tag. Keep it for now maybe?

> - a tag pointing at the first commit of the branch (vYY.N)

Author: Not sure if we still want this ? It doesn't hurt anything, but maybe we should let the first point release create the first tag ?

> Python packages are not uploaded by this process, but are instead created by creating a point release, where the first point release should be `vYY.N.0`.

Reviewer: ```suggestion
Python packages are not published by this process, but are instead published by creating point releases where the first point release should be `vYY.N.0`.
```
Is this better?
